### DISCUSSION TITLE
Stuck on Preparing...

### DISCUSSION THREAD

**Discussion by BerezkovN** [emscripten-core/emscripten#18098] [Q&A] [answered]
Hello! 
I have a simple OpenGL code(from learnopengl) that I build with emcmake. The build is successful.

Though running html file with emrun ends with this:

![image](https://user-images.githubusercontent.com/45544538/197641366-29b007ce-49fa-4576-8d6a-2ded66e43b0e.png)
 
---
Is there any way to debug the cause of this issue?

**Comment by BerezkovN** [comment]
P.S. It's completely stuck. I can't even access console with F12.

**Comment by sbc100** [comment, marked as answer]
Most likely you are not returning from you main function.  With emscripten you need to break up your main loop so that the browser event loop can run.  See https://emscripten.org/docs/porting/emscripten-runtime-environment.html#browser-main-loop

**Reply by BerezkovN** [reply]
Thanks!
